lib/mix/lib/mix/tasks/test.ex

```diff
@@ -642,6 +674,34 @@ defmodule Mix.Tasks.Test do
     Enum.filter(matched_test_files, &MapSet.member?(allowed_files, Path.expand(&1)))
   end
 
+  defp filter_by_partition(files, shell, opts) do
+    if total = opts[:partitions] do
+      partition = System.get_env("MIX_TEST_PARTITION")
+
+      case partition && Integer.parse(partition) do
+        {partition, ""} when partition in 1..total ->
+          partition = partition - 1
+
+          # We sort the files because Path.wildcard does not guarantee
+          # ordering, so different OSes could return a different order,
+          # meaning run across OSes on different partitions could run
+          # duplicate files.
+          for {file, index} <- Enum.with_index(Enum.sort(files)),
```

Comment:
Just to be sure, `Enum.sort/1` isn't using system locale when comparing strings.

Reply:
Yes, it isn't, but I think we can assume we are running on similar file systems, so that the sorting is consistent.

Reply:
It uses `<=` or equivalent semantics.